webapp/src/oEmbed.jsx

```diff
@@ -0,0 +1,65 @@
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
+
+import {id as pluginId} from './manifest';
+import {getDimensionsForProvider, getOEmbedUrl, getProviderForUrl} from './utils';
+
+export default class OEmbed extends Component {
+    static propTypes = {
+        embed: PropTypes.shape({
+            url: PropTypes.string.isRequired,
+        }),
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.oembedUrl = getOEmbedUrl(this.props.embed.url);
+        const provider = getProviderForUrl(this.props.embed.url);
+        this.dimensions = getDimensionsForProvider(provider.provider_name);
+        this.state = {
+            html: null,
+            thumbnailUrl: null,
+        };
+    }
+
+    async componentWillMount() {
```

Comment:
`componentWillMount` is deprecated and will be removed in React 17. This should be replaced with `componentDidMount`